src/java.base/share/classes/java/util/concurrent/SynchronousQueue.java

```diff
@@ -98,718 +101,144 @@ public class SynchronousQueue<E> extends AbstractQueue<E>
-                        while ((item = s.item) == e) {
-                            if ((timed &&
-                                 (nanos = deadline - System.nanoTime()) <= 0) ||
-                                w.isInterrupted()) {
-                                if (s.tryCancel(e)) {
-                                    clean(t, s);
-                                    return null;
-                                }
-                            } else if ((item = s.item) != e) {
-                                break;                   // recheck
-                            } else if (stat <= 0) {
-                                if (t.next == s) {
-                                    if (stat < 0 && t.isFulfilled()) {
-                                        stat = 0;        // yield once if first
-                                        Thread.yield();
-                                    }
-                                    else {
-                                        stat = 1;
-                                        s.waiter = w;
-                                    }
-                                }
-                            } else if (!timed) {
-                                LockSupport.setCurrentBlocker(this);
-                                try {
-                                    ForkJoinPool.managedBlock(s);
-                                } catch (InterruptedException cannotHappen) { }
-                                LockSupport.setCurrentBlocker(null);
-                            }
-                            else if (nanos > SPIN_FOR_TIMEOUT_THRESHOLD)
-                                LockSupport.parkNanos(this, nanos);
-                        }
-                        if (stat == 1)
-                            s.forgetWaiter();
-                        if (!s.isOffList()) {            // not already unlinked
-                            advanceHead(t, s);           // unlink if head
-                            if (item != null)            // and forget fields
-                                s.item = s;
-                        }
-                        return (item != null) ? (E)item : e;
-                    }
-
-                } else if ((m = h.next) != null && t == tail && h == head) {
-                    Thread waiter;
-                    Object x = m.item;
-                    boolean fulfilled = ((isData == (x == null)) &&
-                                         x != m && m.casItem(x, e));
-                    advanceHead(h, m);                    // (help) dequeue
-                    if (fulfilled) {
-                        if ((waiter = m.waiter) != null)
-                            LockSupport.unpark(waiter);
-                        return (x != null) ? (E)x : e;
-                    }
-                }
-            }
-        }
+    /**
+     * The transferer. (See below about serialization.)
+     */
+    private transient final Transferer<E> transferer;
```

Comment:
let's use blessed modifiers order
```suggestion
    private final transient Transferer<E> transferer;
```

Reply:
Thanks. Done